.gitlab-ci.yml

```diff
@@ -0,0 +1,118 @@
+# see https://docs.gitlab.com/ce/ci/yaml/README.html for all available options
+
+variables:
+  SCHEDULER_PARAMETERS: "-J ArborX_CI -W 0:30 -nnodes 1 -P CSC333 -alloc_flags gpumps"
+
+stages:
+  - buildDependencies
+  - buildArborX
+  - runBenchmarkBranch
+  - runBenchmarkMaster
+  - compare
+
+BuildBoost:
+  stage: buildDependencies
+  script:
+    - module load gcc/7.4.0 cuda cmake git
+    - BOOST_VERSION=1.67.0 &&
+      BOOST_VERSION_UNDERSCORE=$(echo "$BOOST_VERSION" | sed -e "s/\./_/g") &&
+      BOOST_URL=https://dl.bintray.com/boostorg/release/${BOOST_VERSION}/source &&
+      BOOST_ARCHIVE=boost_${BOOST_VERSION_UNDERSCORE}.tar.bz2 &&
+      wget --quiet ${BOOST_URL}/${BOOST_ARCHIVE} &&
+      mkdir -p boost &&
+      tar -xf ${BOOST_ARCHIVE} -C boost --strip-components=1 &&
+      cd boost &&
+      ./bootstrap.sh --prefix=${CI_PROJECT_DIR}/boost.install &&
+      ./b2 -j8 hardcode-dll-paths=true dll-path=${CI_PROJECT_DIR}/boost.install link=static variant=release cxxflags=-w install
+  tags:
+    - nobatch
+  artifacts:
+    paths:
+      - boost.install/
+
+BuildBenchmark:
+  stage: buildDependencies
+  script:
+    - module load gcc/7.4.0 cuda cmake git
+    - git clone https://github.com/google/benchmark.git -b v1.4.1 &&
```

Comment:
Why not use a release?

Reply:
I used what we have in our `Dockerfile`.

Reply:
A fair point. We should change it there too.

Reply:
We are checking out a specific tag in the repo

Reply:
Sure, but we don't verify it in any way, not even check the shasum.

Reply:
What about we cache all the dependencies instead?

Reply:
If you can cache, do it. But it does not fix the fact of the original download. But don't focus on this right now, this is a minor nitpick.